test/libmpv_test.c

```diff
@@ -15,15 +15,251 @@
  * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <inttypes.h>
 #include <libmpv/client.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
 
-int main(void)
+// Dummy values for the test.
+const char *str = "string";
+int flag = 1;
+int64_t int_ = 20;
+double double_ = 1.5;
+
+// Global handle.
+mpv_handle *ctx;
+
+static inline void fail(void)
+{
+    mpv_destroy(ctx);
+    exit(1);
+}
+
+static inline void check_api_error(int status)
+{
+    if (status < 0) {
+        printf("mpv API error: %s\n", mpv_error_string(status));
+        fail();
+    }
+}
+
+static inline void check_double(const char *property)
+{
+    double result_double;
+    check_api_error(mpv_get_property(ctx, property, MPV_FORMAT_DOUBLE, &result_double));
+    if (double_ != result_double) {
+        printf("Double: expected '%f' but got '%f'!\n", double_, result_double);
+        fail();
+    }
+}
+
+static inline void check_flag(const char *property)
+{
+    int result_flag;
+    check_api_error(mpv_get_property(ctx, property, MPV_FORMAT_FLAG, &result_flag));
+    if (flag != result_flag) {
+        printf("Flag: expected '%d' but got '%d'!\n", flag, result_flag);
+        fail();
+    }
+}
+
+static inline void check_int(const char *property)
+{
+    int64_t result_int;
+    check_api_error(mpv_get_property(ctx, property, MPV_FORMAT_INT64, &result_int));
+    if (int_ != result_int) {
+        printf("Int: expected '%" PRId64 "' but got '%" PRId64 "'!\n", int_, result_int);
+        fail();
+    }
+}
+
+static inline void check_string(const char *property)
+{
+    char *result_string;
+    check_api_error(mpv_get_property(ctx, property, MPV_FORMAT_STRING, &result_string));
+    if (strcmp(str, result_string) != 0) {
+        printf("Expected '%s' but got '%s'!\n", str, result_string);
+        mpv_free(result_string);
+        fail();
+    }
+    mpv_free(result_string);
+}
+
+static inline void check_results(const char *properties[], enum mpv_format formats[])
```

Comment:
nit: optimizing compilers already do some inline profitability analysis and should be capable of figuring out whether something is profitable to inline or not. And these functions are all `static` so the compiler can see every call-site (unless a function pointer "escapes" out of the TU, which doesn't seem to be happening here).

I'd drop the `inline` noise.

Reply:
Yeah, I can't remember why I put inline here (originally wrote this months ago).